### PR TITLE
Render one frame before showing the window to avoid flashing

### LINKED ISSUE #590: White flash on app startup on Windows 11 dark mode
Xilem apps have a white flash on Windows even on dark mode. This does not happen on macOS. I suspect this has something to do with winit.

Hardware:
OS: Windows 11 23H2 OS Build 22631.4169
CPU: Intel 12600k
GPU: Nvidia RTX 3080 Driver 560.70

This also happens in [Vello examples](https://github.com/linebender/vello/tree/main/examples/with_winit).

Reference [Zulipchat](https://xi.zulipchat.com/#narrow/stream/354396-xilem/topic/Startup.20white.20background.20flash.20on.20Windows.20dark.20mode)

[xilem_calc_white_flash_windows.webm](https://github.com/user-attachments/assets/5af55497-c882-41ef-acfc-6dfc7263204f)

Edit: 
Calc build in video shown is a release build.

### PR DESCRIPTION
**Pull request by Long0x0**
Fixes #590.

Demo:
capture

### CODE REVIEW

**Review by jaredoconnell — COMMENTED** [review]
I can confirm that this works great on Windows 10. I'll let the other reviewers review the code itself.

File: masonry/src/event_loop_runner.rs [hunk 1]
```diff
@@ -280,12 +279,18 @@ impl MasonryState<'_> {
...
-                    window,
+                    window: window.clone(),
                     surface,
                     accesskit_adapter: adapter,
                 };
                 self.render_root
                     .handle_window_event(WindowEvent::Rescale(scale_factor));
+                // Render one frame before showing the window to avoid flashing
+                if visible {
+                    let (scene, _tree_update) = self.render_root.redraw();
```
Comment (DJMcNab) [inline masonry/src/event_loop_runner.rs:290]: Discarding this tree update feels likely to be very bad.

Comment (Long0x0) [reply masonry/src/event_loop_runner.rs]: Yeah, ignoring the tree update could lead to issues. I’ve updated the code to include it, following the approach used in handling the `RedrawRequested` event. Let me know if it looks good now.

**Comment by DJMcNab** [comment]
I'm not set up to test accessibility on my machine yet. But this looks right to me now. I'm treating this similarly to #767, where it's a targeted fix which shouldn't lead to regressions.

**Review by DJMcNab — APPROVED** [review]

**Pull request merged**